src/test/java/uk/gov/hmcts/reform/sendletter/logging/AppInsightsTest.java

```diff
@@ -49,10 +61,77 @@ public void should_fail_creating_app_insights_when_instrumentation_key_is_not_pr
 
     @Test
     public void should_create_app_insights_with_default_dev_mode_off() {
-        context.setInstrumentationKey("some-key");
+        context.setInstrumentationKey(IKEY);
 
         new AppInsights(telemetry);
 
         assertThat(TelemetryConfiguration.getActive().getChannel().isDeveloperMode()).isFalse();
     }
+
+    @Test
+    public void should_track_message_acknowledgement_event_for_success_case() {
+        context.setInstrumentationKey(IKEY);
+
+        AppInsights insights = new AppInsights(telemetry);
+        Map<String, String> properties = Collections.singletonMap("messageId", MESSAGE_ID);
```

Comment:
we can use `ImmutableMap.of("messageId", MESSAGE_ID);` here

Reply:
@nitinprabhu `singletonMap` returns an immutable map as well, no need

Reply:
@rkondratowicz Reason I thought is because at most of the places Immutable map is used it's only at this place we are using singleton.

Reply:
don't see the point of using different library's implementation when it is available in java itself. just a singleton map 😃